packages/nuxt/src/pages/utils.ts

```diff
@@ -241,10 +243,18 @@ export async function getRouteMeta (contents: string, absolutePath: string): Pro
       enter (node) {
         if (foundMeta) { return }
 
-        if (node.type !== 'ExpressionStatement' || node.expression.type !== 'CallExpression' || node.expression.callee.type !== 'Identifier' || node.expression.callee.name !== 'definePageMeta') { return }
+        const class = node.type !== "ArrowFunctionExpression" || node.body.type !== "CallExpression" || node.body.callee.type !== "Identifier" || node.body.callee.name !== "definePageMeta"
+        const other = node.type !== 'ExpressionStatement' || node.expression.type !== 'CallExpression' || node.expression.callee.type !== 'Identifier' || node.expression.callee.name !== 'definePageMeta'
+
+        if (class || other) { return }
 
         foundMeta = true
-        const pageMetaArgument = ((node as ExpressionStatement).expression as CallExpression).arguments[0] as ObjectExpression
+
+        if (class) {
+          const pageMetaArgument = (node as ExpressionStatement).body.arguments[0] as ObjectExpression
+        } else {
+          const pageMetaArgument = ((node as ExpressionStatement).expression as CallExpression).arguments[0] as ObjectExpression
+        }
```

Comment:
this code isn't doing anything, right?

it probably needs a ternary instead:

`const pageMetaArgument = className ? ... : ...`